### PR TITLE
Update SGD optimizer API docs 📓

### PR DESCRIPTION
**Pull request by 8bitmp3**
Good old SGD:

- Description: Added a sentence about momentum and the Nesterov acceleration gradient (N. momentum) being of first-order, etc and what they do to SGD.
- Parameters: Added 4 params with most descriptions from code comments.
- Reference: Added 5 papers (SGD, classic momentum, NAG), such as Robbins and Monro (1951), Wolfowitz (1952), Kiefer and Wolfowitz (1952), Nesterov (Nesterov momentum/NAG, 1983), and Polyak (momentum, 1964).
- Example code snippet: Transfer learning with VGG19 and SGD.

Continuing with the rest of the docs.

Thank you.

### CODE REVIEW

File: Sources/TensorFlow/Optimizers/SGD.swift [hunk 1]
```diff
@@ -14,8 +14,73 @@
...
+/// https://projecteuclid.org/euclid.aoms/1177729586) (Robbins and Monro, 1951), 
+/// ["On the Stochastic Approximation Method of Robbins and Monro"](
+/// https://projecteuclid.org/euclid.aoms/1177729391) (Wolfowitz, 1952) and
+/// ["Stochastic Estimation of the Maximum of a Regression Function"](
+/// https://projecteuclid.org/euclid.aoms/1177729392) (Kiefer and Wolfowitz, 1952), 
+/// ["Some methods of speeding up the convergence of iteration method"](
+/// https://vsokolov.org/courses/750/2018/files/polyak64.pdf) (Polyak, 1964),
+/// ["A method for unconstrained convex minimization problem with the rate of 
+/// convergence"](http://mpawankumar.info/teaching/cdt-big-data/nesterov83.pdf) 
+/// (Nesterov, 1983).
```
Comment (rahulbhalley) [inline Sources/TensorFlow/Optimizers/SGD.swift]: Wow. 🤩 These references look great. 👍

**Comment by dabrahams** [comment]
@dabrahams to schedule meeting with @dan-zheng about this

**Comment by texasmichelle** [comment]
Reminder to @dabrahams and @dan-zheng for follow-up discussion

File: Sources/TensorFlow/Optimizers/SGD.swift [hunk 2]
```diff
@@ -14,8 +14,73 @@
...
+///     - nesterov: A boolean. The Nesterov’s accelerated gradient method. 
+///     (default value: true).
+///
+/// ### Examples: ###
+///  
+/// // - Transfer learning from pre-trained VGG19 weights:
+/// 
+/// ````
+/// ...
+/// // Define the base model architecture of VGG19.
```
Comment (pschuh) [inline Sources/TensorFlow/Optimizers/SGD.swift:50]: Hi, We really like your improvement of the docs, but we find that the example makes the docs too long. We've been thinking of pushing an update to extract the example as an executable example, but haven't found time to get to it. If you're interested in doing this work, that would be great. Thanks.
